modules/miner/miner.go

```diff
@@ -28,12 +34,13 @@ type Miner struct {
 
 	subscribers []chan struct{}
 
-	mu sync.RWMutex
+	mu sync.Mutex
```

Comment:
Interesting. I thought it would be the other way around. I guess it depends how much readlocking you're doing.